### PR TITLE
Add isolation consideration

### PR DESCRIPTION
**Pull request by Nathan219**
Adding the checks for isolation to the elastic and direct urls

### CODE REVIEW

File: lib/runnable-hostname.js [hunk 1]
```diff
@@ -3,22 +3,33 @@
...
 
 module.exports = {
   elastic: elastic,
   direct:  direct
 };
 
+/**
+ * Returns true if the instance is isolated, but is not the isolation master
+ * @param opts
+ */
```
Comment (anandkumarpatel) [inline lib/runnable-hostname.js:19]: missing @returns

File: lib/runnable-hostname.js [hunk 2]
```diff
@@ -3,22 +3,33 @@
...
+  // If the instance is isolated,
+  var replacement = isIsolatedContainer(opts) ? regexForIsolationPrefix : opts.shortHash + '-';
+  return direct(opts).replace(replacement, '').toLowerCase();
 }
 
 function direct (opts) {
   validateOpts(opts);
   var template = '{shortHash}-{repoName}-{env}-{ownerUsername}'
-    .replace('{shortHash}', opts.shortHash)
+    .replace('{shortHash}-', isIsolatedContainer(opts) ? '' : opts.shortHash + '-')
```
Comment (anandkumarpatel) [inline lib/runnable-hostname.js:35]: before this was just `opts.shortHash` but now you are making it `''` if this is an isolated container?

Comment (Nathan219) [inline lib/runnable-hostname.js:35]: Correct, because then it will be {masterInstanceShortHash}--instanceName

**Comment by anandkumarpatel** [comment]
LGTM

**Pull request merged**